tensorboard/webapp/widgets/line_chart_v2/lib/scale.ts

```diff
@@ -0,0 +1,196 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {scaleLinear, scaleLog} from 'd3';
+
+import {ScaleType} from './scale_types';
+
+export interface Scale {
```

Comment:
Does the API require that `domain[1] > domain[0]` or `domain[1] >= domain[0]`?
Must `domain[i]` be finite, or at least non-NaN? Same for `range`.

My intuition is that it might make sense to require

  - all values finite (and thus non-NaN)
  - `domain[1] >= domain[0]`
  - `range[1] > range[0]`

since this is what the calling code can reasonably guarantee: i.e., it’s
filtered NaNs/infs from the dataset and is painting into a non-empty
region. The scales have to nicely handle the case where the domain is
empty/singleton.

What do you think?